tempesta_fw/sched/tfw_sched_http.c

```diff
@@ -322,38 +362,63 @@ tfw_sched_http_cfg_handle_match(TfwCfgSpec *cs, TfwCfgEntry *e)
 	return 0;
 }
 
+static void
+tfw_cfgop_replace_active_rules(TfwHttpMatchList *new_rules)
+{
+	TfwHttpMatchList *active_rules = tfw_rules;
+
+	rcu_assign_pointer(tfw_rules, new_rules);
+	synchronize_rcu();
+
+	tfw_http_match_list_free(active_rules);
+}
+
 /**
  * Delete all rules parsed out of the "sched_http_rules" section.
  */
 static void
-tfw_sched_http_cfg_clean_rules(TfwCfgSpec *cs)
+tfw_cfgop_cleanup_rules(TfwCfgSpec *cs)
 {
-	tfw_http_match_list_free(tfw_sched_http_rules);
-	tfw_sched_http_rules = NULL;
+	tfw_http_match_list_free(tfw_rules_reconfig);
+	tfw_rules_reconfig = NULL;
+
+	if (tfw_runstate_is_reconfig())
```

Comment:
If I correctly understand - this code will be executed after reconfiguration is failed (on parse stage). And in this case Tempesta should continue working with old configuration, but we release old `tfw_rules` here. Maybe condition should be inverted...

Reply:
Yes, you're absolutely right, i'll fix that